### ISSUE TITLE
Filterable tab switcher

### ISSUE THREAD

**Issue by failable** [zed-industries/zed#10145]
### Check for existing issues

- [X] Completed

### Describe the feature

Add search  / filter ability to tab switcher.

### If applicable, add mockups / screenshots to help present your vision of the feature

Zed:
<img width="709" alt="image" src="https://github.com/zed-industries/zed/assets/5475262/2aeaf291-ed5e-4437-98bb-a5ee3292ae3e">

Emacs:
<img width="425" alt="image" src="https://github.com/zed-industries/zed/assets/5475262/826f18a3-774e-46cb-8e67-481e5f9b85a5">
<img width="409" alt="image" src="https://github.com/zed-industries/zed/assets/5475262/d8688268-3643-4bec-8a92-a5845ce83401">

**Comment by mrnugget** [comment]
Since normal `cmd-p` already shows open buffers at the top, I'm wondering whether just showing that a file is open in `Cmd-p` would also help here. Or do you think this is specifically about only seeing the open buffers?

**Comment by failable** [comment]
> I'm wondering whether just showing that a file is open in Cmd-p would also help here. 

Yes, I just notice 😅 that there's a horizontal line separating the open buffers and other workspaces files. If the status is shown obviously enough, maybe this FR seems a bit duplicated.

> Or do you think this is specifically about only seeing the open buffers?

But anyway if FR is implemented, it would still helpful for someone who just ever want to see open buffers.

**Comment by derekperkins** [comment]
if you have a lot of tabs open, it's still nice to be able to filter

**Comment by hron** [comment]
Just for documenting purposes. I think having a shortcut as the following is the perfect solution. Basically, it opens the file finder with the history on top. This is not exactly  filterable tab switcher, but from my perspective it covers many use cases.

```json
{
  {
    "context": "Workspace",
    "bindings": {
      "ctrl-b": ["file_finder::Toggle", { "separate_history": true }]
    }
}
```
![image](https://github.com/zed-industries/zed/assets/39293/3dec2f75-4ee9-46f2-bd9f-e382a7940074)

**Comment by pckilgore** [comment]
> Since normal cmd-p already shows open buffers at the top, I'm wondering whether just showing that a file is open in Cmd-p would also help here. Or do you think this is specifically about only seeing the open buffers?

For me cmd-p will show but not filter such files.  

I would also prefer a command that only filters on open buffers.  I work in a codebase with many similarly named files, and there's a meaningful difference between a switching to the file I know I opened already (of which there is one), vs seeing both that result any the many other files that are in the project.  

The mental difference between "there was one match that's it go" probably looks to someone over my shoulder like the window didn't even flash long enough for me to read it.  Which is true, I didn't, I just saw one thing there.  When there's many results, you have to read them to check if the right one is selected.  To my brain, many results = I made a typo when I filtered.  Since this buffer search feature exists across FZF and telescope for nvim, I doubt I'm the only one!

**Comment by olejorgenb** [comment]
`tab switcher: toggle all` allow you to filter/search for tabs now  (perhaps not the most intuitive name,)

<img width="494" height="431" alt="Image" src="https://github.com/user-attachments/assets/5545d811-812e-4b88-9f81-799763e5f9f4" />
